### DISCUSSION TITLE
ESP32-WROVER Port with I2S DAC?

### DISCUSSION THREAD

**Discussion by NonaSuomy** [divingmule/plugin.audio.iheart#9] [Ideas]
https://github.com/divingmule/plugin.audio.iheart/issues/6#issuecomment-791959032

> Haha, over the past year or so I've been getting into electronics. Mostly simple circuits with micro-controllers. I love it but the math gets over my head really quick :-\
> 
> I've made some good progress this morning. If you'd like to test swap out the default.py with this one >
> [default.zip](https://github.com/divingmule/plugin.audio.iheart/files/6095449/default.zip)

Any chance with your new-found electronics love could you port this to the ESP32-WROVER (Extra PSRAM for streaming)?

Would love this to be a standalone tiny microcontroller streamer. Maybe you are already working on something like this?

I helped add iHeart to Edzelf's platform with Arduino a long time ago. It was simple XMLScraping. They changed the way they did things and it's broken now would love a more dedicated codebase to iHeart with the ESP32 and maybe a more compatible any I2S DAC sound processor build.

https://github.com/Edzelf/ESP32-Radio

https://github.com/schreibfaul1/ESP32-audioI2S

https://github.com/schreibfaul1/ESP32-MiniWebRadio I liked this one because it used generic I2S DAC and an alarm clock unfortunately it's really string heavy.
^- I ported this one to this board https://www.aliexpress.com/item/32846510254.html?spm=2114.12010612.8148356.20.14e636fc5E7IWO

https://github.com/CelliesProjects/wm8978-esp32

https://github.com/CelliesProjects/eStreamPlayer32

https://github.com/karawin/Ka-Radio32 https://www.youtube.com/watch?v=Jk8d02b-uso (LILYGO® TTGO Tm Music Albums 2.4 Inch PCM5102A)

https://github.com/RalphBacon/205-Internet-Radio https://www.youtube.com/watch?v=xrR8EZh2bMI This guy did his own build with some youtube videos.

https://www.aliexpress.com/item/4001095751970.html?spm=a2g0o.cart.0.0.75963c00sig1Wk&mp=1

**Comment by divingmule** [comment]
While this would be a very cool project I don't think it's something I would pursue. My experience building GUI's is minimal, much less on a micro-controller. The learning curve, time investment is greater than my interest. 

That being said the code is open for anyone who would like to pursue it and I would be happy to help on the back-end side of things.

**Reply by NonaSuomy** [reply]
Any chance you can explain the basics of how you pull the station streams from the unpublished API's and play the streams you receive with KODI.

Be nice if I could just get it working again with this codebase https://github.com/Edzelf/ESP32-Radio/blob/0c4268192b276f8981e5c808ec4cdb1c14ba843a/src/main.cpp#L3835 this is how I was doing it before.

Used to do this shown above in the url and basic below:

```
  const char* xmlhost = "playerservices.streamtheworld.com" ;  // XML data source
  const char* xmlget =  "GET /api/livestream"                  // XML get parameters
                        "?version=1.5"                         // API Version of IHeartRadio
                        "&mount=%sAAC"                  // MountPoint with Station Callsign
                        "&lang=en" ;                           // Language

http://playerservices.streamtheworld.com/api/livestream?version=1.5&mount=IHR_TRANAAC&lang=en
```
